pkg/cli/image/extract/extract.go

```diff
@@ -357,7 +360,13 @@ func (o *ExtractOptions) Run() error {
 					return err
 				}
 
-				imageConfig, layers, err := imagemanifest.ManifestToImageConfig(ctx, srcManifest, repo.Blobs(ctx), location)
+				manifests, err := repo.Manifests(ctx)
+				if err != nil {
+					return err
+				}
+
+				// If empty, set platformSpec to match other mappings
+				imageConfig, layers, platformSpec, err = imagemanifest.ManifestToImageConfig(ctx, srcManifest, repo.Blobs(ctx), manifests, location, true, platformSpec)
```

Comment:
TODO: have to wire in FilterByOS if `oc image extract` should work w/ --filter-by-os.  Set platformSpec to that, and then this PR will fix both cmds.